core-bundle/src/EventListener/DataContainer/LogoutPageRedirectListener.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCallback;
+use Contao\DataContainer;
+use Contao\Module;
+
+#[AsCallback(table: 'tl_page', target: 'fields.jumpTo.attributes')]
+class LogoutPageRedirectListener
+{
+    /**
+     * @param DataContainer|Module|null $dc
+     */
+    public function __invoke(array $attributes, $dc): array
```

Comment:
```suggestion
    public function __invoke(array $attributes, object|null $dc): array
```

If we cannot narrow it down to `DataContainer|Module|null`, we should at least use `object|null` instead of no type-hint at all.

Reply:
We don't know who is using `Widget:: getAttributesFromDca()` and passing what value. I don't think our callback can accept a narrower selection than the widget method.

Reply:
If those type-hints were enforced, the method would only accept `DataContainer|Module|null` anyway. What does Haste pass?

Reply:
A mock `DataContainer` in the latest version apparently: https://github.com/codefog/contao-haste/blob/main/src/Form/Form.php#L430

But I'm not sure what we're discussing here. We can either not have a type hint at all (use `mixed`) or keep it as-is. We cannot enforce this without certainly breaking stuff for no reason.